From 0087b7f1698eca2ea1a1503e5e1d4ed33d3bc50d Mon Sep 17 00:00:00 2001
From: timor <timor.dd@googlemail.com>
Date: Sun, 4 Jul 2021 11:04:55 +0200
Subject: [PATCH] Normalize "x-" variants for comparing protocol and mime part
 types

---
 gmime/gmime-multipart-signed.c | 39 ++++++++++++++++++++++++++++++++--
 1 file changed, 37 insertions(+), 2 deletions(-)

diff --git a/gmime/gmime-multipart-signed.c b/gmime/gmime-multipart-signed.c
index f157ed71..c24b8fc7 100644
--- a/gmime/gmime-multipart-signed.c
+++ b/gmime/gmime-multipart-signed.c
@@ -352,6 +352,40 @@ check_protocol_supported (const char *protocol, const char *supported)
 	return rv;
 }
 
+static gboolean
+check_protocol_match (const char *protocol, const char * mime_type)
+{
+	const char *subtype;
+	const char *protocol_subtype;
+	char *xsupported;
+	gboolean rv;
+
+	if (!protocol)
+		return FALSE;
+
+	if (!g_ascii_strcasecmp (protocol, mime_type))
+		return TRUE;
+
+	if (!(subtype = strrchr (mime_type, '/')))
+		return FALSE;
+	subtype++;
+
+	if (!(protocol_subtype = strrchr (protocol, '/')))
+		return FALSE;
+	protocol_subtype++;
+
+	if (!g_ascii_strncasecmp (protocol_subtype, "x-", 2))
+		protocol_subtype += 2;
+	if (!g_ascii_strncasecmp (subtype, "x-", 2))
+		subtype += 2;
+
+	if (!g_ascii_strcasecmp (protocol_subtype, subtype))
+		return TRUE;
+
+	return FALSE;
+}
+
+
 
 /**
  * g_mime_multipart_signed_verify:
@@ -415,10 +449,11 @@ g_mime_multipart_signed_verify (GMimeMultipartSigned *mps, GMimeVerifyFlags flag
 	}
 	
 	signature = g_mime_multipart_get_part ((GMimeMultipart *) mps, GMIME_MULTIPART_SIGNED_SIGNATURE);
-	
+
 	/* make sure the protocol matches the signature content-type */
 	mime_type = g_mime_content_type_get_mime_type (signature->content_type);
-	if (g_ascii_strcasecmp (mime_type, protocol) != 0) {
+
+	if (!check_protocol_match (protocol, mime_type)) {
 		g_set_error_literal (err, GMIME_ERROR, GMIME_ERROR_PARSE_ERROR,
 				     _("Cannot verify multipart/signed part: signature content-type does not match protocol."));
 		g_object_unref (ctx);
-- 
2.29.3

