--- plasma-workspace-5.16.5/lookandfeel/contents/lockscreen/LockScreenUi.qml	2019-09-03 15:59:19.000000000 +0200
+++ plasma-workspace-5.16.5/lookandfeel/contents/lockscreen/LockScreenUi.qml.patched	2020-03-21 15:52:01.388388064 +0100
@@ -36,6 +36,19 @@
 
     colorGroup: PlasmaCore.Theme.ComplementaryColorGroup
 
+    function performOperation(what) {
+        var service = dataEngine.serviceForSource("PowerDevil");
+        var operation = service.operationDescription(what);
+        service.startOperationCall(operation);
+    }
+
+    PlasmaCore.DataSource {
+      id: dataEngine
+      engine: "powermanagement"
+      connectedSources: ["PowerDevil"]
+    }
+
+
     Connections {
         target: authenticator
         onFailed: {
@@ -112,6 +125,7 @@
             }
         }
         Keys.onPressed: {
+            if (event.key == 16908292) performOperation("suspendToRam")
             uiVisible = true;
             event.accepted = false;
         }
@@ -252,6 +266,11 @@
                         onClicked: mainStack.push(switchSessionPage)
                         // the current session isn't listed in the model, hence a check for greater than zero, not one
                         visible: (sessionsModel.count > 0 || sessionsModel.canStartNewSession) && sessionsModel.canSwitchUser
+                    },
+                    ActionButton {
+                      text: i18nd("plasma_lookandfeel_org.kde.lookandfeel", "Suspend")
+                      iconSource: "system-suspend"
+                      onClicked: performOperation("suspendToRam")
                     }
                 ]
 
