diff -Nru emacs-25.3/lisp/erc/erc-compat.el emacs-25.3-new/lisp/erc/erc-compat.el
--- emacs-25.3/lisp/erc/erc-compat.el	2017-04-14 17:02:47.000000000 +0200
+++ emacs-25.3-new/lisp/erc/erc-compat.el	2017-09-16 17:53:07.093316346 +0200
@@ -66,7 +66,7 @@
 	 user-emacs-directory)
 	((boundp 'user-init-directory)
 	 user-init-directory)
-	(t "~/.emacs.d/"))
+	(t "~/.spacemacs.d/"))
   "Directory beneath which additional per-user Emacs-specific files
 are placed.
 Note that this should end with a directory separator.")
diff -Nru emacs-25.3/lisp/progmodes/js.el emacs-25.3-new/lisp/progmodes/js.el
--- emacs-25.3/lisp/progmodes/js.el	2017-04-14 17:02:47.000000000 +0200
+++ emacs-25.3-new/lisp/progmodes/js.el	2017-09-16 18:07:45.153578684 +0200
@@ -497,7 +497,7 @@
   :group 'js)
 
 (defcustom js-js-tmpdir
-  "~/.emacs.d/js/js"
+  "~/.spacemacs.d/js/js"
   "Temporary directory used by `js-mode' to communicate with Mozilla.
 This directory must be readable and writable by both Mozilla and Emacs."
   :type 'directory
diff -Nru emacs-25.3/lisp/startup.el emacs-25.3-new/lisp/startup.el
--- emacs-25.3/lisp/startup.el	2017-04-14 17:02:47.000000000 +0200
+++ emacs-25.3-new/lisp/startup.el	2017-09-16 19:00:32.881761209 +0200
@@ -1192,16 +1192,16 @@
 		      ;; This tells `load' to store the file name found
 		      ;; into user-init-file.
 		      (setq user-init-file t)
-		      (load user-init-file-1 t t)
+		      ;; (load user-init-file-1 t t) inhibit user init file loading completely!
 
-		      (when (eq user-init-file t)
+		      (when t           ;since we inhibited loading .emacs, always load ./spacemacs.d.d/init.el
 			;; If we did not find ~/.emacs, try
 			;; ~/.emacs.d/init.el.
 			(let ((otherfile
 			       (expand-file-name
 				"init"
 				(file-name-as-directory
-				 (concat "~" init-file-user "/.emacs.d")))))
+				 (concat "~" init-file-user "/.spacemacs.d.d")))))
 			  (load otherfile t t)
 
 			  ;; If we did not find the user's init file,
diff -Nru emacs-25.3/lisp/subr.el emacs-25.3-new/lisp/subr.el
--- emacs-25.3/lisp/subr.el	2017-04-14 17:02:47.000000000 +0200
+++ emacs-25.3-new/lisp/subr.el	2017-09-16 17:53:05.696339247 +0200
@@ -2638,8 +2638,8 @@
 (defconst user-emacs-directory
   (if (eq system-type 'ms-dos)
       ;; MS-DOS cannot have initial dot.
-      "~/_emacs.d/"
-    "~/.emacs.d/")
+      "~/_spacemacs.d.d/"
+    "~/.spacemacs.d.d/")
   "Directory beneath which additional per-user Emacs-specific files are placed.
 Various programs in Emacs store information in this directory.
 Note that this should end with a directory separator.
diff -Nru emacs-25.3/src/xterm.c emacs-25.3-new/src/xterm.c
--- emacs-25.3/src/xterm.c	2017-04-14 17:02:47.000000000 +0200
+++ emacs-25.3-new/src/xterm.c	2017-09-16 17:53:08.238297575 +0200
@@ -11896,7 +11896,7 @@
 #if ! GTK_CHECK_VERSION (2, 90, 0)
         /* Load our own gtkrc if it exists.  */
         {
-          const char *file = "~/.emacs.d/gtkrc";
+          const char *file = "~/.spacemacs.d/gtkrc";
           Lisp_Object s, abs_file;
 
           s = build_string (file);
