diff --git a/basis/editors/emacs/emacs.factor b/basis/editors/emacs/emacs.factor
index f2d2240cf7..a565bfb54a 100644
--- a/basis/editors/emacs/emacs.factor
+++ b/basis/editors/emacs/emacs.factor
@@ -19,11 +19,14 @@ M: windows find-emacsclient
         [ "emacsclient.exe" ]
     } 0|| ;
 
+SYMBOL: emacsclient-args
+
 M: emacsclient editor-command ( file line -- command )
     [
         emacsclient-path get [ find-emacsclient ] unless* ,
         "-a=emacs" ,
         "--no-wait" ,
+        emacsclient-args get %
         number>string "+" prepend ,
         ,
     ] { } make ;
